arithmetization/src/main/java/net/consensys/linea/corset/RustCorsetValidator.java

```diff
@@ -0,0 +1,228 @@
+    }
+    // success!
+    return new Result(true, traceFile.toFile(), outcome.output());
+  }
+
+  /**
+   * Attempt to locate the corset executable. Ideally, this is on the PATH and works out-of-the-box.
+   * However, this will fall back to a default location (based on where Cargo normally places
+   * binaries) if that is not the case.
+   */
+  private void initCorset() {
+    // Try default
+    if (isExecutable("corset", "--version")) {
+      corsetBin = "corset";
+      active = true;
+    } else {
+      // Try fall back
+      final String homePath = System.getenv("HOME");
+      corsetBin = homePath + "/.cargo/bin/corset";
+      log.warn("Trying to use default corset path: %s".formatted(corsetBin));
+      // Sanity check we can execute it
+      active = isExecutable(corsetBin, "--version");
+    }
+  }
+
+  /**
+   * Configure corset from the <code>CORSET_FLAGS</code> environment variable (if set). If the
+   * environment variable is not set, the default configuration is retained. If the environment
+   * variable is set, but its value is malformed then an exception is raised.
+   */
+  private void configCorset() {
+    String flags = System.getenv().get("CORSET_FLAGS");
+    if (flags != null) {
+      log.info(
+          "Configuring corset from CORSET_FLAGS environment variable: \"%s\"".formatted(flags));
+      // Reset default configuration
+      this.fieldArithmetic = false;
+      this.expansion = 0;
+      this.autoConstraints = false;
+      // Check for default case (empty string)
+      if (!flags.isEmpty()) {
+        // split flags by separator
+        String[] splitFlags = flags.split(",");
+        // Build configuration based on flags
+        for (String flag : splitFlags) {
+          switch (flag) {
+            case "fields":
+              this.fieldArithmetic = true;
+              break;
+            case "expand":
+              if (expansion >= 4) {
+                throw new RuntimeException(
+                    ("Malformed Corset configuration flags (expansion "
+                            + "beyond four meaningless): %s")
+                        .formatted(flags));
+              }
+              this.expansion++;
+              break;
+            case "auto":
+              this.autoConstraints = true;
+              break;
+            default:
+              // Error
+              throw new RuntimeException("Unknown Corset configuration flag: %s".formatted(flag));
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Construct the command-line to use for running corset to check a given trace file is accepted
+   * (or not) by a given bin file. Amongst other things, this will configure the command-line flags
+   * as dictated by the CORSET_FLAGS environment variable.
+   *
+   * @return
+   */
+  private List<String> buildCommandLine(Path traceFile, String zkEvmBin) {
+    ArrayList<String> options = new ArrayList<>();
```

Comment:
Please use the interface type here `List<String>`.

Reply:
Ok, done.  Also rebased to HEAD for `arith-dev`.